File: InteractiveLive/Android/VideoOne/component/ToolKit/src/main/java/com/vertcdemo/core/eventbus/SolutionEventBus.java
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0

package com.vertcdemo.core.eventbus;

import android.util.Log;

import org.greenrobot.eventbus.EventBus;

public class SolutionEventBus {
    private static final String TAG = "SolutionBus";

    private static final EventBus sInstance = EventBus.getDefault();

    public static void post(Object object) {
        Log.d(TAG, "event=" + object.getClass());
        sInstance.post(object);
    }

    public static void register(Object object) {
        sInstance.register(object);
    }

    public static void unregister(Object object) {
        sInstance.unregister(object);
    }
}

File: Client/Android/component/SolutionBase/src/main/java/com/vertcdemo/core/eventbus/SolutionEventBus.kt
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0
package com.vertcdemo.core.eventbus

import android.util.Log
import org.greenrobot.eventbus.EventBus

object SolutionEventBus {
    private const val TAG = "SolutionBus"
    private val eventBus = EventBus.getDefault()

    @JvmStatic
    fun post(event: Any) {
        Log.d(TAG, "event=${event.javaClass}")
        eventBus.post(event)
    }

    @JvmStatic
    fun register(subscriber: Any) {
        eventBus.register(subscriber)
    }

    @JvmStatic
    fun unregister(subscriber: Any) {
        eventBus.unregister(subscriber)
    }
}
